dask/tests/test_order.py

```diff
@@ -243,3 +225,23 @@ def test_order_doesnt_fail_on_mixed_type_keys():
     order({'x': (inc, 1),
            ('y', 0): (inc, 2),
            'z': (add, 'x', ('y', 0))})
+
+
+def test_gh_3055():
+    da = pytest.importorskip('dask.array')
+    A, B = 20, 99
+    x = da.random.normal(size=(A, B), chunks=(1, None))
+    for _ in range(2):
+        y = (x[:, None, :] * x[:, :, None]).cumsum(axis=0)
+        x = x.cumsum(axis=0)
+    w = (y * x[:, None]).sum(axis=(1,2))
+
+    dsk = dict(w.__dask_graph__())
+    o = order(dsk)
+    L = [o[k] for k in w.__dask_keys__()]
+    assert sorted(L) == L or sorted(L) == L[::-1]
```

Comment:
I assume this is due to the random behavior above? In either case, I'd rather this was a consistent test result, as it'll make it easier to debug in the future.

Reply:
Not related to the randomization above, I just didn't care what order it came in as long as it was sorted.  Happy to take off the second piece.